File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/non-plagiarized/04/T07.py
import numpy as np

def sum_major_diagonal(matrix):
    return np.trace(matrix)

def main():
    print("Enter a 4-by-4 matrix row by row")
    matrix = np.array([list(map(float, input().split())) for _ in range(4)])
    print(f"Sum of the elements in the major diagonal is {sum_major_diagonal(matrix)}")

if __name__ == "__main__":
    main()

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/non-plagiarized/02/T07.ts
import * as readline from 'readline';

/**
 *
 * @author 65FBEF05E01FAC390CB3FA073FB3E8CF (452BF208BF901322968557227B8F6EFE010)
 */
function main(): void {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    console.log("Enter a 4-by-4 matrix row by row:");

    rl.on('line', (input: string) => {
        const m: number[][] = [];
        const rows = input.trim().split('\n');
        for (const row of rows) {
            m.push(row.trim().split(' ').map(Number));
        }

        console.log(`Sum of the elements in the major diagonal is ${sumMajorDiagonal(m)}`);
        rl.close();
    });
}

function sumMajorDiagonal(m: number[][]): number {
    let temp = 0;
    for (let i = 0; i < 4; i++) {
        for (let j = 0; j < 4; j++) {
            if (i === j) {
                temp += m[i][j];
            }
        }
    }
    return temp;
}

main();

